### PR TITLE
Update documentation for Tokens, Session, and Native and Web App Integration

### PR DESCRIPTION
**Pull request by coldlink**
## What does this change?

Updates our native apps integration guide in order to be better up to date with how to go about implementing sign in, registration, and reset/set password for native iOS and Android apps.

Adds documentation on how to integrate web applications with Okta, based on the preliminary research done for the spike. This document should be finalised once the outcomes from the spike have been decided.

Also adds information on the difference between all the OAuth tokens, and adds documentation on the different meaning of the word "sessions" when it comes to Okta.

### CODE REVIEW

File: docs/okta/native-apps-registration.md [hunk 1]
```diff
@@ -1,59 +0,0 @@
-### Apps registration with Okta
```
Comment (coldlink) [inline docs/okta/native-apps-registration.md:1]: This wasn't relevant any more, and has already been included in the other file.

File: docs/okta/native-apps-integration-guide.md [hunk 2]
```diff
@@ -6,21 +6,53 @@ To hand control back to the App Native layer from the In-App Browser Tab we reco
...
 
-To initiate login or registration the app with use the SDK to call the relevant method. In Android this is done by calling `.signIn(...)` and in iOS by calling `.signInWithBrowser(...)`, without any additional parameters. The SDK uses this method to launch the Authorization Code Flow with PKCE. It will launch an in-app browser tab (iOS: `ASWebAuthenticationSession`, Android: `CustomTabsService` (Custom Tab)) to check if a user session exists in Okta.
+To initiate login or registration the app with use the SDK to call the relevant method. In Android this is done by calling `.signIn(...)` and in iOS by calling `.signInWithBrowser(...)`, without any additional parameters. The SDK uses this method to launch the Authorization Code Flow with PKCE. It will launch an in-app browser tab (iOS: `ASWebAuthenticationSession`, Android: `CustomTabsService` (Custom Tab)) to check if a user session exists in Okta. In most cases you will want to pass the `prompt=login` parameter to the SDK to ensure that the user is shown the sign in page regardless of if they have an existing session or not.
```
Comment (raphaelkabo) [inline docs/okta/native-apps-integration-guide.md:15]: 'will use the SDK', I think?

File: docs/okta/sessions.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Application Session
+
+We generally call this the OAuth Session, Local Session, or App Session internally, but is known in the Okta documentation as the Application Session.
+
+The application session is identified through the use of OAuth tokens, specifically the Access and ID tokens, which are created when an app goes through a given OAuth flow, in most cases this will be the Authorization Code flow (with or without PKCE).
+
+When creating an application session, the OAuth flow will check if the user has an IdP session, if they do the user is authenticated using that IdP session, and the OAuth flow completes and creates an application session. If a user does not have an IdP session, the OAuth flow will ask the user to first authenticate with their credentials, therefore creating an IdP session, and then the OAuth flow will complete and create an application session.
+
+Application sessions will have a much shorter session length than the IdP session, usually in the range of 1 hour or a few hours. In order to extend the given application session, the user must go through the OAuth flow again, which will refresh the application session (and the tokens associated with it).
```
Comment (raphaelkabo) [inline docs/okta/sessions.md:38]: Do we actually use this/plan to use this approach anywhere? (I.e. an approach without refresh tokens, where app sessions expire rapidly). It sounds from this like we do.

**Review by raphaelkabo — APPROVED** [review]
Incredible work, well done!

**Pull request merged**